examples/tests/TokenLinker.js

```diff
@@ -0,0 +1,193 @@
+'use strict';
+
+const chai = require('chai');
+const {
+    utils: { defaultAbiCoder, keccak256, id },
+    constants: { AddressZero },
+} = require('ethers');
+const { deployCreate3Upgradable } = require('@axelar-network/axelar-gmp-sdk-solidity/index');
+
+const { expect } = chai;
+const { ethers } = require('hardhat');
+
+const TokenLinkerProxy = require('../../artifacts/examples/evm/token-linker/TokenLinkerProxy.sol/TokenLinkerProxy.json');
+const TokenLinkerLockUnlock = require('../../artifacts/examples/evm/token-linker/TokenLinkerLockUnlock.sol/TokenLinkerLockUnlock.json');
+const TokenLinkerMintBurn = require('../../artifacts/examples/evm/token-linker/TokenLinkerMintBurn.sol/TokenLinkerMintBurn.json');
+const TokenLinkerNative = require('../../artifacts/examples/evm/token-linker/TokenLinkerNative.sol/TokenLinkerNative.json');
+
+const getRandomID = () => id(Math.floor(Math.random() * 1e10).toString());
+
+describe('TokenLinker', () => {
+    let gatewayFactory;
+    let gasServiceFactory;
+    let tokenFactory;
+    let create3DeployerFactory;
+
+    let gateway;
+    let gasService;
+    let tokenLinker;
+    let token;
+    let create3Deployer;
+
+    let ownerWallet;
+    let userWallet;
+
+    const sourceChain = 'chainA';
+    const destinationChain = 'chainB';
+    const tokenName = 'testToken';
+    const tokenSymbol = 'TEST';
+    const decimals = 16;
+
+    const approve = (payloadHash, commandId, txHash, txIndex) => {
+        const approveData = defaultAbiCoder.encode(
+            ['string', 'string', 'address', 'bytes32', 'bytes32', 'uint256'],
+            [sourceChain, tokenLinker.address, tokenLinker.address, payloadHash, txHash, txIndex],
+        );
+
+        return gateway.approveContractCall(approveData, commandId);
+    };
+
+    before(async () => {
+        [ownerWallet, userWallet] = await ethers.getSigners();
+
+        gatewayFactory = await ethers.getContractFactory('MockGateway', ownerWallet);
+        gasServiceFactory = await ethers.getContractFactory('MockGasService', ownerWallet);
+        tokenFactory = await ethers.getContractFactory('ERC20MintableBurnable', ownerWallet);
+        create3DeployerFactory = await ethers.getContractFactory('Create3Deployer', ownerWallet);
```

Comment:
why is create3 deployer even needed? the test is deploying only a contract on a single chain. create2/create3 is needed only when deploying with the same address on multiple chains. in fact, create3 will fail if salt is reused for the same chain